client/src/components/HomePage/Registration.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+import { Input, Button, Row } from 'react-materialize'
+import { emailValidation, request } from '../../utils';
+
+class Registration extends Component {
+  state = {
+    email: '',
+    nickname: '',
+    password: '',
+    repeat_password: '',
+    msg: '',
+    check_email: false,
+    check_nickname: false,
+    check_password: false
+  };
+
+  onChangeHandler = (e) => {
+    let { id } = e.currentTarget
+    this.setState({ [id]: e.currentTarget.value })
+  };
+
+  register = (e) => {
+    let registerData = {
+      email: this.state.email,
+      nickname: this.state.nickname,
+      password: this.state.password,
+      repeat_password: this.state.repeat_password,
+    }
+    if (
+      !emailValidation(this.state.email)
+      || !this.state.password.length
+      || this.state.password !== this.state.repeat_password
+    ) {
+      this.setState({ check_password: "passwords isn't equal" })
+      window.Materialize.toast("Invalid input", 3000)
+      return null
+    }
+    request('/registration', "POST", JSON.stringify(registerData))
+      .then(data => {
+        this.setState({ msg: data.msg })
+        if (
+          this.state.msg === "Your email address is already registered"
```

Comment:
Nope. If you need to determine what field is wrong, return this info fro backend instead of comparing error messages, which can be changed in the future.